lib/puppet/type/cisco_interface.rb

```diff
@@ -473,6 +494,205 @@ def insync?(is)
     newvalues(:true, :false, :default)
   end # property vlan_mapping_enable
 
+  ############################
+  # spanning-tree attributes #
+  ############################
+
+  newproperty(:stp_bpdufilter) do
+    desc 'Enable/Disable BPDU filtering for this interface.'
+
+    newvalues(:enable, :disable, :default)
+  end # property stp_bpdufilter
+
+  newproperty(:stp_bpduguard) do
+    desc 'Enable/Disable BPDU guard for this interface.'
+
+    newvalues(:enable, :disable, :default)
+  end # property stp_bpduguard
+
+  newproperty(:stp_cost) do
+    desc "Spanning tree port path cost for this interface. Valid values are
+          integer, keyword 'auto' or 'default'."
+
+    munge do |value|
+      value = :default if value == 'default'
+      value = :auto if value == 'auto'
+      begin
+        value = Integer(value) unless value == :default || value == :auto
+      rescue
+        raise 'stp_cost must be a valid integer, or auto or default.'
+      end
+      value
+    end
+  end # property stp_cost
+
+  newproperty(:stp_guard) do
+    desc 'Spanning-tree guard mode for this interface.'
+
+    newvalues(:loop, :none, :root, :default)
+  end # property stp_guard
+
+  newproperty(:stp_link_type) do
+    desc 'Link type for spanning tree tree protocol use.'
+
+    newvalues(:auto, :shared, :'point-to-point', :default)
+  end # property stp_link_type
+
+  # rubocop:disable Style/PredicateName
```

Comment:
This is because of the Puppet magic `is_to_s` method, right? Instead of disabling it on a per-file basis like this, can you update the .rubocop.yml file with the following:

```yaml
Style/PredicateName:
  NameWhiteList:
    - is_a?
    - is_to_s
```

This should keep Rubocop from complaining about `is_to_s` across all files in the module.